Heracles reports execution fails for Google BigQuery

Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 400 Bad Request
{
 "code" : 400,
 "errors" : [ {
   "domain" : "global",
   "location" : "q",
   "locationType" : "parameter",
   "message" : "Unrecognized name: istinct at [3:32]",
   "reason" : "invalidQuery"
 } ],
 "message" : "Unrecognized name: istinct at [3:32]",
 "status" : "INVALID_ARGUMENT"
}
2019-06-21 18:59:53.825 ERROR taskExecutor-5 org.springframework.batch.core.step.AbstractStep -  - Encountered an error executing step cohortAnalysisStep in job cohortAnalysisJob
org.springframework.jdbc.UncategorizedSQLException: StatementCallback; uncategorized SQLException for SQL []; SQL state [HY000]; error code [100033]; [Simba][BigQueryJDBCDriver](100033) Error getting job status.; nested exception is java.sql.SQLException: [Simba][BigQueryJDBCDriver](100033) Error getting job status.

Currently the issue is under research if it's environment problem

--- a/src/main/resources/resources/cohortresults/sql/cohortSpecific/ageAtIndex.sql
+++ b/src/main/resources/resources/cohortresults/sql/cohortSpecific/ageAtIndex.sql
@@ -1,4 +1,4 @@
-select cast(stratum_1 as integer) as age_at_index,
+select CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) as age_at_index,
 	count_value as num_persons
 from @ohdsi_database_schema.heracles_results
 where analysis_id in (1800)

--- a/src/main/resources/resources/cohortresults/sql/cohortSpecific/byConcept/drugOccursRelativeToIndex.sql
+++ b/src/main/resources/resources/cohortresults/sql/cohortSpecific/byConcept/drugOccursRelativeToIndex.sql
@@ -9,33 +9,33 @@ SELECT hr1.cohort_definition_id,
          ELSE 0 
        END     AS pct_persons 
 FROM   (SELECT cohort_definition_id, 
-               Cast(stratum_1 AS INTEGER)      AS concept_id, 
-               Cast(stratum_2 AS INTEGER) * 30 AS duration, 
+               CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)      AS concept_id,
+               CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) * 30 AS duration,
                count_value 
         FROM   @ohdsi_database_schema.heracles_results 
         WHERE  analysis_id IN ( 1870 ) 
                AND cohort_definition_id = @cohortDefinitionId) hr1 
        INNER JOIN (SELECT cohort_definition_id, 
-                          -1 * Cast(stratum_1 AS INTEGER) * 30                AS 
+                          -1 * CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) * 30                AS
               duration, 
                           Sum(count_value) 
                             OVER ( 
                               partition BY cohort_definition_id 
-                              ORDER BY -1* Cast(stratum_1 AS INTEGER)*30 ASC) AS 
+                              ORDER BY -1* CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)*30 ASC) AS
               count_value 
                    FROM   @ohdsi_database_schema.heracles_results 
                    WHERE  analysis_id IN ( 1805 ) 
                           AND cohort_definition_id = @cohortDefinitionId 
-                          AND Cast(stratum_1 AS INTEGER) > 0 
+                          AND CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) > 0
                    UNION 
-                   SELECT hr1.cohort_definition_id, 
-                          Cast(hr1.stratum_1 AS INTEGER) * 30 
+                   SELECT hr1.cohort_definition_id,
+                          CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT) * 30
                           AS 
                           duration, 
                           t1.count_value - Sum(hr1.count_value) 
                           OVER ( 
                             partition BY hr1.cohort_definition_id 
-                            ORDER BY Cast(hr1.stratum_1 AS INTEGER)* 
+                            ORDER BY CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT)*
                           30 ASC) AS 
                           count_value 
                    FROM   @ohdsi_database_schema.heracles_results hr1 
@@ -53,13 +53,13 @@ FROM   (SELECT cohort_definition_id,
                ON hr1.cohort_definition_id = t1.cohort_definition_id 
                   AND hr1.duration = t1.duration 
        INNER JOIN (SELECT cohort_definition_id, 
-                          Cast(stratum_1 AS INTEGER) AS concept_id, 
+                          CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) AS concept_id,
                           Sum(count_value)           AS count_value 
                    FROM   @ohdsi_database_schema.heracles_results 
                    WHERE  analysis_id IN ( 1870 ) 
                           AND cohort_definition_id = @cohortDefinitionId 
                    GROUP  BY cohort_definition_id, 
-                             Cast(stratum_1 AS INTEGER) 
+                             CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)
                    HAVING Sum(heracles_results.count_value) > @minCovariatePersonCount) ct1
                ON hr1.cohort_definition_id = ct1.cohort_definition_id 
                   AND hr1.concept_id = ct1.concept_id 
@@ -79,33 +79,33 @@ SELECT hr1.cohort_definition_id,
          ELSE 0 
        END   AS pct_persons 
 FROM   (SELECT cohort_definition_id, 
-               Cast(stratum_1 AS INTEGER)      AS concept_id, 
-               Cast(stratum_2 AS INTEGER) * 30 AS duration, 
+               CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)      AS concept_id,
+               CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) * 30 AS duration,
                count_value 
         FROM   @ohdsi_database_schema.heracles_results 
         WHERE  analysis_id IN ( 1871 ) 
                AND cohort_definition_id = @cohortDefinitionId) hr1 
        INNER JOIN (SELECT cohort_definition_id, 
-                          -1 * Cast(stratum_1 AS INTEGER) * 30                AS 
+                          -1 * CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) * 30                AS
               duration, 
                           Sum(count_value) 
                             OVER ( 
                               partition BY cohort_definition_id 
-                              ORDER BY -1* Cast(stratum_1 AS INTEGER)*30 ASC) AS 
+                              ORDER BY -1* CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)*30 ASC) AS
               count_value 
                    FROM   @ohdsi_database_schema.heracles_results 
                    WHERE  analysis_id IN ( 1805 ) 
                           AND cohort_definition_id = @cohortDefinitionId 
-                          AND Cast(stratum_1 AS INTEGER) > 0 
+                          AND CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) > 0
                    UNION 
                    SELECT hr1.cohort_definition_id, 
-                          Cast(hr1.stratum_1 AS INTEGER) * 30 
+                         CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT) * 30
                           AS 
                           duration, 
                           t1.count_value - Sum(hr1.count_value) 
                           OVER ( 
                             partition BY hr1.cohort_definition_id 
-                            ORDER BY Cast(hr1.stratum_1 AS INTEGER)* 
+                            ORDER BY CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT)*
                           30 ASC) AS 
                           count_value 
                    FROM   @ohdsi_database_schema.heracles_results hr1 
@@ -123,13 +123,13 @@ FROM   (SELECT cohort_definition_id,
                ON hr1.cohort_definition_id = t1.cohort_definition_id 
                   AND hr1.duration = t1.duration 
        INNER JOIN (SELECT cohort_definition_id, 
-                          Cast(stratum_1 AS INTEGER) AS concept_id, 
+                          CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) AS concept_id,
                           Sum(count_value)           AS count_value 
                    FROM   @ohdsi_database_schema.heracles_results 
                    WHERE  analysis_id IN ( 1871 ) 
                           AND cohort_definition_id = @cohortDefinitionId 
                    GROUP  BY cohort_definition_id, 
-                             Cast(stratum_1 AS INTEGER) 
+                             CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)
                    HAVING Sum(heracles_results.count_value) > @minCovariatePersonCount) ct1
                ON hr1.cohort_definition_id = ct1.cohort_definition_id 
                   AND hr1.concept_id = ct1.concept_id 

--- a/src/main/resources/resources/cohortresults/sql/cohortSpecific/byConcept/firstConditionRelativeToIndex.sql
+++ b/src/main/resources/resources/cohortresults/sql/cohortSpecific/byConcept/firstConditionRelativeToIndex.sql
@@ -9,19 +9,19 @@ SELECT hr1.cohort_definition_id,
          ELSE 0 
        END     AS pct_persons 
 FROM   (SELECT cohort_definition_id, 
-               Cast(stratum_1 AS INTEGER)      AS concept_id, 
-               Cast(stratum_2 AS INTEGER) * 30 AS duration, 
+               CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)      AS concept_id,
+               CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) * 30 AS duration,
                count_value 
         FROM   @ohdsi_database_schema.heracles_results 
         WHERE  analysis_id IN ( 1820 ) 
                AND cohort_definition_id = @cohortDefinitionId
-               AND Cast(stratum_2 AS INTEGER) * 30 BETWEEN -1000 AND 1000
+               AND CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) * 30 BETWEEN -1000 AND 1000
                         ) hr1 
-       INNER JOIN (SELECT -1 * Cast(stratum_1 AS INTEGER) * 30                AS 
+       INNER JOIN (SELECT -1 * CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) * 30                AS
               duration, 
                           Sum(count_value) 
                             OVER ( 
-                              ORDER BY -1* Cast(stratum_1 AS INTEGER)*30 ASC) AS 
+                              ORDER BY -1* CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)*30 ASC) AS
               count_value 
                    FROM  (
                                              select stratum_1, max(count_value) as count_value
@@ -34,7 +34,7 @@ FROM   (SELECT cohort_definition_id,
 
                                                   union
 
-                                                  select cast(stratum_1 as integer) as stratum_1, count_value
+                                                  select CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) as stratum_1, count_value
                                                   from @ohdsi_database_schema.heracles_results
                                                   where analysis_id = 1805
                                                   and cohort_definition_id = @cohortDefinitionId
@@ -44,15 +44,15 @@ FROM   (SELECT cohort_definition_id,
                                              ) t0
                                              
                                              
-                                             WHERE Cast(stratum_1 AS INTEGER) > 0 
+                                             WHERE CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) > 0
                    UNION 
-                   SELECT Cast(hr1.stratum_1 AS INTEGER) * 30 
+                   SELECT CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT) * 30
                           AS 
                           duration, 
                           t1.count_value - Sum(hr1.count_value) 
                           OVER ( 
                             partition BY hr1.cohort_definition_id 
-                            ORDER BY Cast(hr1.stratum_1 AS INTEGER)* 
+                            ORDER BY CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT)*
                           30 ASC) AS 
                           count_value 
                    FROM   @ohdsi_database_schema.heracles_results hr1 
@@ -69,11 +69,11 @@ FROM   (SELECT cohort_definition_id,
                           AND hr1.cohort_definition_id = @cohortDefinitionId
                                                     ) t1 
                ON  hr1.duration = t1.duration 
-       INNER JOIN (SELECT Cast(stratum_1 AS INTEGER) AS concept_id, 
+       INNER JOIN (SELECT CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) AS concept_id,
                           Sum(count_value)           AS count_value 
                    FROM   @ohdsi_database_schema.heracles_results 
                    WHERE  analysis_id IN ( 1820 ) 
-                           GROUP  BY Cast(stratum_1 AS INTEGER) 
+                           GROUP  BY CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)
 										HAVING Sum(heracles_results.count_value) > @minCovariatePersonCount
                    ) ct1 
                ON hr1.concept_id = ct1.concept_id 
@@ -92,19 +92,19 @@ SELECT hr1.cohort_definition_id,
          ELSE 0 
        END   AS pct_persons 
 FROM   (SELECT cohort_definition_id, 
-               Cast(stratum_1 AS INTEGER)      AS concept_id, 
-               Cast(stratum_2 AS INTEGER) * 30 AS duration, 
+               CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)      AS concept_id,
+               CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) * 30 AS duration,
                count_value 
         FROM   @ohdsi_database_schema.heracles_results  
         WHERE  analysis_id IN ( 1821 ) 
                AND cohort_definition_id = @cohortDefinitionId
-               AND Cast(stratum_2 AS INTEGER) * 30 BETWEEN -1000 AND 1000
+               AND CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) * 30 BETWEEN -1000 AND 1000
                ) hr1 
-       INNER JOIN (SELECT -1 * Cast(stratum_1 AS INTEGER) * 30                AS 
+       INNER JOIN (SELECT -1 * CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) * 30                AS
               duration, 
                           Sum(count_value) 
                             OVER ( 
-                              ORDER BY -1* Cast(stratum_1 AS INTEGER)*30 ASC) AS 
+                              ORDER BY -1* CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)*30 ASC) AS
               count_value 
                    FROM  (
                                              select stratum_1, max(count_value) as count_value
@@ -117,7 +117,7 @@ FROM   (SELECT cohort_definition_id,
 
                                                   union
 
-                                                  select cast(stratum_1 as integer) as stratum_1, count_value
+                                                  select CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) as stratum_1, count_value
                                                   from @ohdsi_database_schema.heracles_results
                                                   where analysis_id = 1805
                                                   and cohort_definition_id = @cohortDefinitionId
@@ -127,15 +127,15 @@ FROM   (SELECT cohort_definition_id,
                                              ) t0
                                              
                                              
-                                             WHERE Cast(stratum_1 AS INTEGER) > 0 
+                                             WHERE CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) > 0
                    UNION 
-                   SELECT Cast(hr1.stratum_1 AS INTEGER) * 30 
+                   SELECT CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT) * 30
                           AS 
                           duration, 
                           t1.count_value - Sum(hr1.count_value) 
                           OVER ( 
                             partition BY hr1.cohort_definition_id 
-                            ORDER BY Cast(hr1.stratum_1 AS INTEGER)* 
+                            ORDER BY CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT)*
                           30 ASC) AS 
                           count_value 
                    FROM   @ohdsi_database_schema.heracles_results hr1 
@@ -153,11 +153,11 @@ FROM   (SELECT cohort_definition_id,
                               ( @cohortDefinitionId )
                                                     ) t1 
                ON  hr1.duration = t1.duration 
-       INNER JOIN (SELECT Cast(stratum_1 AS INTEGER) AS concept_id, 
+       INNER JOIN (SELECT CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) AS concept_id,
                           Sum(count_value)           AS count_value 
                    FROM   @ohdsi_database_schema.heracles_results 
                    WHERE  analysis_id IN ( 1820 ) 
-                           GROUP  BY Cast(stratum_1 AS INTEGER) 
+                           GROUP  BY CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)
 										HAVING Sum(heracles_results.count_value) > @minCovariatePersonCount
                    ) ct1 
                ON hr1.concept_id = ct1.concept_id 

--- a/src/main/resources/resources/cohortresults/sql/cohortSpecific/byConcept/firstConditionRelativeToIndex.sql
+++ b/src/main/resources/resources/cohortresults/sql/cohortSpecific/byConcept/firstConditionRelativeToIndex.sql
@@ -17,11 +17,11 @@ FROM   (SELECT cohort_definition_id,
                AND cohort_definition_id = @cohortDefinitionId
                AND CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) * 30 BETWEEN -1000 AND 1000
                         ) hr1 
-       INNER JOIN (SELECT -1 * CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) * 30                AS
+       INNER JOIN (SELECT -1 * CAST(stratum_1 AS INT) * 30                AS
               duration, 
                           Sum(count_value) 
                             OVER ( 
-                              ORDER BY -1* CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)*30 ASC) AS
+                              ORDER BY -1* CAST(stratum_1 AS INT)*30 ASC) AS
               count_value 
                    FROM  (
                                              select stratum_1, max(count_value) as count_value
@@ -44,7 +44,7 @@ FROM   (SELECT cohort_definition_id,
                                              ) t0
                                              
                                              
-                                             WHERE CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) > 0
+                                             WHERE CAST(stratum_1 AS INT) > 0
                    UNION 
                    SELECT CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT) * 30
                           AS 
@@ -100,11 +100,11 @@ FROM   (SELECT cohort_definition_id,
                AND cohort_definition_id = @cohortDefinitionId
                AND CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) * 30 BETWEEN -1000 AND 1000
                ) hr1 
-       INNER JOIN (SELECT -1 * CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) * 30                AS
+       INNER JOIN (SELECT -1 * CAST(stratum_1 AS INT) * 30                AS
               duration, 
                           Sum(count_value) 
                             OVER ( 
-                              ORDER BY -1* CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)*30 ASC) AS
+                              ORDER BY -1* CAST(stratum_1 AS INT)*30 ASC) AS
               count_value 
                    FROM  (
                                              select stratum_1, max(count_value) as count_value
@@ -127,7 +127,7 @@ FROM   (SELECT cohort_definition_id,
                                              ) t0
                                              
                                              
-                                             WHERE CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) > 0
+                                             WHERE CAST(stratum_1 AS INT) > 0
                    UNION 
                    SELECT CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT) * 30
                           AS 

--- a/src/main/resources/resources/cohortresults/sql/cohortSpecific/byConcept/procedureOccursRelativeToIndex.sql
+++ b/src/main/resources/resources/cohortresults/sql/cohortSpecific/byConcept/procedureOccursRelativeToIndex.sql
@@ -18,33 +18,33 @@ SELECT hr1.cohort_definition_id,
          ELSE 0 
        END     AS pct_persons 
 FROM   (SELECT cohort_definition_id, 
-               Cast(stratum_1 AS INTEGER)      AS concept_id, 
-               Cast(stratum_2 AS INTEGER) * 30 AS duration, 
+               CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)      AS concept_id,
+               CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) * 30 AS duration,
                count_value 
         FROM   @ohdsi_database_schema.heracles_results 
         WHERE  analysis_id IN ( 1830 ) 
                AND cohort_definition_id = @cohortDefinitionId) hr1 
        INNER JOIN (SELECT cohort_definition_id, 
-                          -1 * Cast(stratum_1 AS INTEGER) * 30                AS 
+                          -1 * CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) * 30                AS
               duration, 
                           Sum(count_value) 
                             OVER ( 
                               partition BY cohort_definition_id 
-                              ORDER BY -1* Cast(stratum_1 AS INTEGER)*30 ASC) AS 
+                              ORDER BY -1* CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)*30 ASC) AS
               count_value 
                    FROM   @ohdsi_database_schema.heracles_results 
                    WHERE  analysis_id IN ( 1805 ) 
                           AND cohort_definition_id = @cohortDefinitionId 
-                          AND Cast(stratum_1 AS INTEGER) > 0 
+                          AND CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) > 0
                    UNION 
                    SELECT hr1.cohort_definition_id, 
-                          Cast(hr1.stratum_1 AS INTEGER) * 30 
+                          CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT) * 30
                           AS 
                           duration, 
                           t1.count_value - Sum(hr1.count_value) 
                           OVER ( 
                             partition BY hr1.cohort_definition_id 
-                            ORDER BY Cast(hr1.stratum_1 AS INTEGER)* 
+                            ORDER BY  CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT)*
                           30 ASC) AS 
                           count_value 
                    FROM   @ohdsi_database_schema.heracles_results hr1 
@@ -63,13 +63,13 @@ FROM   (SELECT cohort_definition_id,
                ON hr1.cohort_definition_id = t1.cohort_definition_id 
                   AND hr1.duration = t1.duration 
        INNER JOIN (SELECT cohort_definition_id, 
-                          Cast(stratum_1 AS INTEGER) AS concept_id, 
+                          CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) AS concept_id,
                           Sum(count_value)           AS count_value 
                    FROM   @ohdsi_database_schema.heracles_results 
                    WHERE  analysis_id IN ( 1830 ) 
                           AND cohort_definition_id = @cohortDefinitionId 
                    GROUP  BY cohort_definition_id, 
-                             Cast(stratum_1 AS INTEGER) 
+                             CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)
                    HAVING Sum(heracles_results.count_value) > @minCovariatePersonCount) ct1
                ON hr1.cohort_definition_id = ct1.cohort_definition_id 
                   AND hr1.concept_id = ct1.concept_id 
@@ -89,33 +89,33 @@ SELECT hr1.cohort_definition_id,
          ELSE 0 
        END   AS pct_persons 
 FROM   (SELECT cohort_definition_id, 
-               Cast(stratum_1 AS INTEGER)      AS concept_id, 
-               Cast(stratum_2 AS INTEGER) * 30 AS duration, 
+               CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)      AS concept_id,
+               CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) * 30 AS duration,
                count_value 
         FROM   @ohdsi_database_schema.heracles_results 
         WHERE  analysis_id IN ( 1831 ) 
                AND cohort_definition_id = @cohortDefinitionId) hr1 
        INNER JOIN (SELECT cohort_definition_id, 
-                          -1 * Cast(stratum_1 AS INTEGER) * 30                AS 
+                          -1 * CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) * 30                AS
               duration, 
                           Sum(count_value) 
                             OVER ( 
                               partition BY cohort_definition_id 
-                              ORDER BY -1* Cast(stratum_1 AS INTEGER)*30 ASC) AS 
+                              ORDER BY -1* CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)*30 ASC) AS
               count_value 
                    FROM   @ohdsi_database_schema.heracles_results 
                    WHERE  analysis_id IN ( 1805 ) 
                           AND cohort_definition_id = @cohortDefinitionId 
-                          AND Cast(stratum_1 AS INTEGER) > 0 
+                          AND CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) > 0
                    UNION 
                    SELECT hr1.cohort_definition_id, 
-                          Cast(hr1.stratum_1 AS INTEGER) * 30 
+                          CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT) * 30
                           AS 
                           duration, 
                           t1.count_value - Sum(hr1.count_value) 
                           OVER ( 
                             partition BY hr1.cohort_definition_id 
-                            ORDER BY Cast(hr1.stratum_1 AS INTEGER)* 
+                            ORDER BY  CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT)*
                           30 ASC) AS 
                           count_value 
                    FROM   @ohdsi_database_schema.heracles_results hr1 
@@ -134,13 +134,13 @@ FROM   (SELECT cohort_definition_id,
                ON hr1.cohort_definition_id = t1.cohort_definition_id 
                   AND hr1.duration = t1.duration 
        INNER JOIN (SELECT cohort_definition_id, 
-                          Cast(stratum_1 AS INTEGER) AS concept_id, 
+                          CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) AS concept_id,
                           Sum(count_value)           AS count_value 
                    FROM   @ohdsi_database_schema.heracles_results 
                    WHERE  analysis_id IN ( 1831 ) 
                           AND cohort_definition_id = @cohortDefinitionId 
                    GROUP  BY cohort_definition_id, 
-                             Cast(stratum_1 AS INTEGER) 
+                            CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)
                    HAVING Sum(heracles_results.count_value) > @minCovariatePersonCount) ct1
                ON hr1.cohort_definition_id = ct1.cohort_definition_id 
                   AND hr1.concept_id = ct1.concept_id 

--- a/src/main/resources/resources/cohortresults/sql/cohortSpecific/conditionOccurrencePrevalenceOfCondition.sql
+++ b/src/main/resources/resources/cohortresults/sql/cohortSpecific/conditionOccurrencePrevalenceOfCondition.sql
@@ -23,8 +23,8 @@ select   concept_hierarchy.concept_id,
 from
 (select stratum_1 as concept_id,
 	sum(count_value) as num_persons,
-	sum(case when CAST(stratum_2 AS INT) < 0 then count_value else 0 end) as num_persons_before,
-	sum(case when CAST(stratum_2 AS INT) > 0 then count_value else 0 end) as num_persons_after
+	sum(case when CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) < 0 then count_value else 0 end) as num_persons_before,
+	sum(case when CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) > 0 then count_value else 0 end) as num_persons_after
 from @ohdsi_database_schema.heracles_results
 where analysis_id in (1820) --first occurrence of condition
 and cohort_definition_id = @cohortDefinitionId

--- a/src/main/resources/resources/cohortresults/sql/cohortSpecific/drugEraPrevalenceOfDrug.sql
+++ b/src/main/resources/resources/cohortresults/sql/cohortSpecific/drugEraPrevalenceOfDrug.sql
@@ -31,8 +31,8 @@ select   concept_hierarchy.concept_id,
 from
 (select stratum_1 as concept_id,
 	sum(count_value) as num_persons,
-	sum(case when CAST(stratum_2 AS INT) < 0 then count_value else 0 end) as num_persons_before,
-	sum(case when CAST(stratum_2 AS INT) > 0 then count_value else 0 end) as num_persons_after
+	sum(case when CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) < 0 then count_value else 0 end) as num_persons_before,
+	sum(case when CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) > 0 then count_value else 0 end) as num_persons_after
 from @ohdsi_database_schema.heracles_results
 where analysis_id in (1870) --first occurrence of drug
 and cohort_definition_id = @cohortDefinitionId

--- a/src/main/resources/resources/cohortresults/sql/cohortSpecific/getTimeToEventDrilldown.sql
+++ b/src/main/resources/resources/cohortresults/sql/cohortSpecific/getTimeToEventDrilldown.sql
@@ -20,16 +20,16 @@ WITH denominator AS (
 
 			SELECT exposure_cohort_definition_id, 
 				outcome_cohort_definition_id,
-				cast(stratum_2 AS INTEGER) as stratum_1, sum(count_value) as count_value
+				CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) as stratum_1, sum(count_value) as count_value
 			FROM @ohdsi_database_schema.penelope_results
 			WHERE exposure_cohort_definition_id = @exposure_cohort_definition_id
 			AND outcome_cohort_definition_id = @outcome_cohort_definition_id
 			AND analysis_id = 1805
 			GROUP BY exposure_cohort_definition_id, 
 				outcome_cohort_definition_id,
-				cast(stratum_2 AS INTEGER)
+				CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT)
 		) t0
-		WHERE CAST(stratum_1 AS INTEGER)<= 10 OR count_value > 0
+		WHERE CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)<= 10 OR count_value > 0
 		GROUP BY exposure_cohort_definition_id, outcome_cohort_definition_id, stratum_1 
 	) t1
 
@@ -57,16 +57,16 @@ WITH denominator AS (
 
 			SELECT exposure_cohort_definition_id, 
 				outcome_cohort_definition_id,
-				cast(stratum_2 AS INTEGER) as stratum_1, sum(count_value) as count_value
+				CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) as stratum_1, sum(count_value) as count_value
 			FROM @ohdsi_database_schema.penelope_results
 			WHERE exposure_cohort_definition_id = @exposure_cohort_definition_id
 			AND outcome_cohort_definition_id = @outcome_cohort_definition_id
 			AND analysis_id = 1806
 			GROUP BY exposure_cohort_definition_id, 
 				outcome_cohort_definition_id,
-				cast(stratum_2 AS INTEGER)
+				CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT)
 		) t0
-		WHERE CAST(stratum_1 AS INTEGER)<= 10 OR count_value > 0
+		WHERE CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)<= 10 OR count_value > 0
 		GROUP BY exposure_cohort_definition_id, outcome_cohort_definition_id, stratum_1 
 	) t1
 	INNER JOIN
@@ -88,7 +88,7 @@ numerator_first as
 (
 	SELECT exposure_cohort_definition_id, 
 		outcome_cohort_definition_id,
-		CAST(stratum_1 AS INTEGER)*30 AS duration, 
+		CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)*30 AS duration,
 		count_value
 	FROM @ohdsi_database_schema.penelope_results
 	WHERE exposure_cohort_definition_id = @exposure_cohort_definition_id
@@ -99,7 +99,7 @@ numerator_all as
 (
 	SELECT exposure_cohort_definition_id, 
 		outcome_cohort_definition_id,
-		CAST(stratum_1 AS INTEGER)*30 AS duration, 
+		CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)*30 AS duration,
 		count_value
 	FROM @ohdsi_database_schema.penelope_results
 	WHERE exposure_cohort_definition_id = @exposure_cohort_definition_id

--- a/src/main/resources/resources/cohortresults/sql/cohortSpecific/observationPeriodTimeRelativeToIndex.sql
+++ b/src/main/resources/resources/cohortresults/sql/cohortSpecific/observationPeriodTimeRelativeToIndex.sql
@@ -6,19 +6,21 @@ select hr1.cohort_definition_id,
 from
 (
 select cohort_definition_id,
-		-1* cast(stratum_1 as integer)*30 as duration,
-		sum(count_value) over (partition by cohort_definition_id order by -1* cast(stratum_1 as integer)*30 asc) as count_value
+		-1* CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)*30 as duration,
+		sum(count_value) over (partition by cohort_definition_id
+		order by -1* CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT)*30 asc) as count_value
 	from
 	@ohdsi_database_schema.heracles_results
 	where analysis_id in (1805)
 	and cohort_definition_id = @cohortDefinitionId
-	and cast(stratum_1 as integer) > 0
+	and CAST(CASE WHEN isNumeric(stratum_1) = 1 THEN stratum_1 ELSE null END AS INT) > 0
 
 	union
 
 	select hr1.cohort_definition_id,
-		cast(hr1.stratum_1 as integer)*30 as duration,
-		t1.count_value - sum(hr1.count_value) over (partition by hr1.cohort_definition_id order by cast(hr1.stratum_1 as integer)*30 asc) as count_value
+	  CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT)*30 as duration,
+		t1.count_value - sum(hr1.count_value) over (partition by hr1.cohort_definition_id
+		order by CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT)*30 asc) as count_value
 	from
 	@ohdsi_database_schema.heracles_results hr1
 	inner join

--- a/src/main/resources/resources/cohortresults/sql/cohortSpecific/personsInCohortFromCohortStartToEnd.sql
+++ b/src/main/resources/resources/cohortresults/sql/cohortSpecific/personsInCohortFromCohortStartToEnd.sql
@@ -1,7 +1,7 @@
 --persons in cohort from cohort start time to cohort end time, in 30d increments
-select cast(hr1.stratum_1 as int) as month_year, 
+select CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT) as month_year,
   hr1.count_value as count_value, 
 round(1.0*hr1.count_value / denom.count_value,5) as percent_value
 from (select * from @ohdsi_database_schema.heracles_results where analysis_id = 1804 and cohort_definition_id = @cohortDefinitionId) hr1,
 (select count_value from @ohdsi_database_schema.heracles_results where analysis_id = 1 and cohort_definition_id = @cohortDefinitionId) denom
-order by hr1.stratum_1 asc
+order by CAST(CASE WHEN isNumeric(hr1.stratum_1) = 1 THEN hr1.stratum_1 ELSE null END AS INT) asc

--- a/src/main/resources/resources/cohortresults/sql/cohortSpecific/prevalenceByYearGenderSex.sql
+++ b/src/main/resources/resources/cohortresults/sql/cohortSpecific/prevalenceByYearGenderSex.sql
@@ -8,20 +8,20 @@ select
 	round(1000*(1.0*hr1.count_value / t1.count_value),5) as Y_PREVALENCE_1000PP
 from (select cohort_definition_id,
 	stratum_1 as index_year,
-	cast(stratum_2 as integer) as gender_concept_id,
-	cast(stratum_3 as integer) as age_decile,
+	CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) as gender_concept_id,
+	CAST(CASE WHEN isNumeric(stratum_3) = 1 THEN stratum_3 ELSE null END AS INT) as age_decile,
 	count_value 
 	from @ohdsi_database_schema.heracles_results
 	where analysis_id in (1814)
 	and cohort_definition_id = @cohortDefinitionId
-	and cast(stratum_2 as integer) in (8507,8532)
-	and cast(stratum_3 as integer) >= 0 
+	and CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) in (8507,8532)
+	and CAST(CASE WHEN isNumeric(stratum_3) = 1 THEN stratum_3 ELSE null END AS INT) >= 0
 ) hr1
 	inner join 
 (
 	select stratum_1 as index_year,
-	cast(stratum_2 as integer) as gender_concept_id,
-	cast(stratum_3 as integer) as age_decile,
+	CAST(CASE WHEN isNumeric(stratum_2) = 1 THEN stratum_2 ELSE null END AS INT) as gender_concept_id,
+	CAST(CASE WHEN isNumeric(stratum_3) = 1 THEN stratum_3 ELSE null END AS INT) as age_decile,
 	count_value 
 	from @ohdsi_database_schema.heracles_results 
 	where analysis_id = 116

